### PR TITLE
CCD-4140: addition of new functionality and test cases

### PR DESCRIPTION
**Pull request by dinesh1patel**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CCD-4140

### Change description ###

Addition of new functionality and test cases

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: bulk-user-setup/bulk-user-setup.config [hunk 1]
```diff
@@ -47,13 +55,25 @@ NORMAL=$(tput sgr0)
 ###########################################
 
 #'|' delimited string of roles. These roles will be added by default for all caseworkers
-DEFAULT_ROLES="caseworker"
+#DEFAULT_ROLES="caseworker"
```
Comment (rajaram-nagarajan) [inline bulk-user-setup/bulk-user-setup.config:58]: We can remove the unused global variables

File: bulk-user-setup/bulk-user-setup.config [hunk 2]
```diff
@@ -47,13 +55,25 @@ NORMAL=$(tput sgr0)
...
+
+IGNORED_ROLES_FROM_USER_ADD_REQUEST="caseworker"
+
+IGNORED_ROLES_FROM_USER_DELETE_REQUEST="caseworker|payments|payments-refund|payments-refund-approver"
+
+#'|' delimited string of roles. These roles will be added by default for all caseworkers
+ADD_ROLES_BY_DEFAULT="caseworker"
 
-DEFAULT_CASEWORKER_ROLE="caseworker"
+#'|' delimited string of roles. These roles will be deleted by default for all caseworkers
```
Comment (rajaram-nagarajan) [inline bulk-user-setup/bulk-user-setup.config:68]: We can change the comment to 
**These roles will be deleted by default for all caseworkers when there is no other caseworker related roles available for that user**

File: bulk-user-setup/bulk-user-setup.config [hunk 3]
```diff
@@ -47,13 +55,25 @@ NORMAL=$(tput sgr0)
...
 #'|' delimited string of roles. These roles will be added by default for all caseworkers
-DEFAULT_ROLES="caseworker"
+#DEFAULT_ROLES="caseworker"
+#DEFAULT_CASEWORKER_ROLE="caseworker"
+
+IGNORED_ROLES_FROM_USER_ADD_REQUEST="caseworker"
+
+IGNORED_ROLES_FROM_USER_DELETE_REQUEST="caseworker|payments|payments-refund|payments-refund-approver"
+
+#'|' delimited string of roles. These roles will be added by default for all caseworkers
```
Comment (rajaram-nagarajan) [inline bulk-user-setup/bulk-user-setup.config:65]: We can change the comment to
**These roles will be added by default for all caseworkers when there are other caseworker related roles available for that user**

File: bulk-user-setup/test/utils/idam-create-users.sh [hunk 4]
```diff
@@ -201,7 +201,8 @@ function update_user() {
 ## MAIN
 ######################
 current_dir=$(pwd)
-source ${current_dir}/./bulk-user-setup.config > out.log 2> /dev/null
+# source ${current_dir}/./bulk-user-setup.config > out.log 2> /dev/null
```
Comment (rajaram-nagarajan) [inline bulk-user-setup/test/utils/idam-create-users.sh:204]: we can remove the unwanted code

File: bulk-user-setup/bulk-user-setup.config (hunk 1 above)
Comment (dinesh1patel) [reply bulk-user-setup/bulk-user-setup.config]: Dones

File: bulk-user-setup/bulk-user-setup.config (hunk 2 above)
Comment (dinesh1patel) [reply bulk-user-setup/bulk-user-setup.config]: Done

File: bulk-user-setup/bulk-user-setup.config (hunk 3 above)
Comment (dinesh1patel) [reply bulk-user-setup/bulk-user-setup.config]: Done

File: bulk-user-setup/test/utils/idam-create-users.sh (hunk 4 above)
Comment (dinesh1patel) [reply bulk-user-setup/test/utils/idam-create-users.sh]: Done

**Review by rajaram-nagarajan — APPROVED** [review]
LGTM

**Pull request merged**